client/components/tinymce/plugins/simple-payments/dialog/product-image-picker.jsx

```diff
@@ -0,0 +1,140 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+
+import AsyncLoad from 'components/async-load';
+import { getMediaItem } from 'state/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import MediaActions from 'lib/media/actions';
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import ProductImage from './product-image';
+import RemoveButton from 'components/remove-button';
+
+class ProductImagePicker extends Component {
+	static propTypes = {
+		featuredImage: PropTypes.object,
+		siteId: PropTypes.number,
+		translate: PropTypes.func,
+	};
+
+	state = {
+		isSelecting: false,
+	};
+
+	showMediaModal = event => {
+		const { siteId, featuredImage } = this.props;
+
+		if ( event.key && event.key !== 'Enter' ) {
+			// A11y - prevent opening Media modal with any key
+			return;
+		}
+
+		this.setState( { isSelecting: true }, () => {
+			if ( featuredImage ) {
+				MediaActions.setLibrarySelectedItems( siteId, [ featuredImage ] );
+			}
+		} );
+	};
+
+	setImage = value => {
+		this.setState( { isSelecting: false }, () => {
+			if ( ! value ) {
+				return;
+			}
+
+			this.props.input.onChange( value.items[ 0 ].ID );
+		} );
+	};
+
+	removeCurrentImage = event => {
+		event.stopPropagation();
+
+		this.props.input.onChange( false );
+	};
+
+	getImagePlaceholder() {
+		return (
+			<div
+				className="dialog__product-image-placeholder"
+				onClick={ this.showMediaModal }
+				onKeyDown={ this.showMediaModal }
+				role="button"
+				tabIndex={ 0 }
+			>
+				<Gridicon icon="add-image" size={ 36 } />
+				{ this.props.translate( 'Add an Image' ) }
+			</div>
+		);
+	}
+
+	getCurrentImage() {
+		const { siteId } = this.props;
+
+		return (
+			<div
+				className="dialog__product-image"
+				onClick={ this.showMediaModal }
+				onKeyDown={ this.showMediaModal }
+				role="button"
+				tabIndex={ 0 }
+			>
+				<ProductImage siteId={ siteId } imageId={ this.props.input.value } showEditIcon={ true } />
```

Comment:
Minor nitpick: we could simplify `showEditIcon={ true }` into `showEditIcon`.

Reply:
Updated in https://github.com/Automattic/wp-calypso/pull/24797 (although for consistency reasons I'm leaning towards always specifying that explicitly).